users/validators.py

```
@@ -16,3 +18,57 @@
+
+    # Find all <a> tags with href attributes
+    link_pattern = r'<a\s+[^>]*href=["\']([^"\']+)["\'][^>]*>'
+    links = re.findall(link_pattern, value, re.IGNORECASE)
+
+    # Check all links are internal
+    for href in links:
+        if not is_internal_url(href):
+            raise ValidationError(
+                f"External links are not allowed. Found: {href}. "
+                f'Only internal links starting with "/" or "#" are permitted.'
+            )
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)